source/PlayerInfo.cpp

```diff
@@ -1356,7 +1370,21 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		auto oldFirstShip = ships[0];
+		int pos = 0;
```

Comment:
since this variable stores a position of a vector element, size_t would be more universal
```suggestion
		size_t pos = 0;
```